### PR TITLE
add loss function; modify Backward

### PR DESCRIPTION
**Pull request by YiwenShaoStephen**
Add a temporary loss function just for denominator to show how we are going to use it in a PyTorch environment. We can add numerator to it when it's ready.

### CODE REVIEW

File: pytorch_binding/src/chain-denominator.cc [hunk 1]
```diff
@@ -297,8 +297,8 @@ BaseFloat DenominatorComputation::ComputeTotLogLike() {
 
 
 
-bool DenominatorComputation::Backward(
-    BaseFloat deriv_weight, at::Tensor nnet_output_deriv) {
+at::Tensor DenominatorComputation::Backward() {
+  at::Tensor nnet_output_deriv = at::zeros_like(exp_nnet_output_transposed_).transpose(0, 1);
```
Comment (vimalmanohar) [inline pytorch_binding/src/chain-denominator.cc:301]: We should not initialize the memory here. That should be done with appropriate sizes before in python itself maybe. We are just adding whatever was there before to this. So don't return Tensor object, it's confusing where the memory needs to be initialized.

Comment (YiwenShaoStephen) [reply pytorch_binding/src/chain-denominator.cc:301]: But it seems to me that initialization memory in C++/CUDA part is a much common case here. As you can see in the example they provide: [https://github.com/pytorch/extension-cpp/blob/master/cuda/lltm_cuda_kernel.cu]. So if in the python code calls the function in C, we will return a new tensor instead of changing the data in the put tensor, which otherwise would be really confusing in python code. Because usually if you want to do such a in-place function, we should imply it by adding suffix '_' to the function name.

Comment (vimalmanohar) [reply pytorch_binding/src/chain-denominator.cc:301]: That is a different scenario. Here we are storing the denominator backward and numerator backward gradients in the same Tensor. So the memory should be allocated before. It should not be zeroed inside the function.

Comment (YiwenShaoStephen) [reply pytorch_binding/src/chain-denominator.cc:301]: Ok, I get it. I will change it back in my next commit.
